build.zig

```diff
@@ -22,11 +22,60 @@ pub fn build(b: *std.Build) !void {
 
     const mime_module = try GenerateMimeTypes.generateMimeModule(b);
 
-    b.installArtifact(lib);
+    const exe = b.addExecutable(.{
+        .name = "jetzig",
+        .root_source_file = .{ .path = "src/main.zig" },
+        .target = target,
+        .optimize = optimize,
+    });
+
+    const zig_args_dep = b.dependency("args", .{ .target = target, .optimize = optimize });
+    exe.root_module.addImport("args", zig_args_dep.module("args"));
+
+    b.installArtifact(exe);
+
+    const install_step = b.getInstallStep();
+    const install_source_step = b.addInstallDirectory(.{
+        .source_dir = .{ .path = "." },
+        .include_extensions = &[_][]const u8{ ".zig", ".zmpl", ".zon", ".png", ".ico", ".css", ".gitignore" },
+        .install_dir = .{ .prefix = {} },
+        .install_subdir = "jetzig",
+    });
+    install_step.dependOn(&install_source_step.step);
+
+    const run_cmd = b.addRunArtifact(exe);
+
+    run_cmd.step.dependOn(b.getInstallStep());
+
+    if (b.args) |args| {
+        run_cmd.addArgs(args);
+    }
+
+    const run_step = b.step("run", "Run the app");
+    run_step.dependOn(&run_cmd.step);
 
     const jetzig_module = b.addModule("jetzig", .{ .root_source_file = .{ .path = "src/jetzig.zig" } });
     jetzig_module.addImport("mime_types", mime_module);
     lib.root_module.addImport("jetzig", jetzig_module);
+    exe.root_module.addImport("jetzig", jetzig_module);
+
+    // TODO: Skip when not running install command
+    var git_commit = b.run(&[_][]const u8{ "git", "rev-parse", "--verify", "HEAD" });
```

Comment:
While this is good, I would use an external program making an http request to the github api, and get the result argument for it

Reply:
If not, we cannot do the following:

```zig
    const jetzig_dep = b.dependency("jetzig", .{.optimize = .ReleaseFast});
    const jetzig_cli = b.addRunArtifact(jetzig_dep.artifact("jetzig"));

    // Run jetzig cli tool
    if(b.args) |args| {
        jetzig_cli.addArgs(args);
    }

    const run_jetzig_cli = b.step("jetzig", "runs the jetzig app");
    run_jetzig_cli.dependOn(run_jetzig_cli);

```

Reply:
init script fetches latest `main` when setting up now.

In future we can store current Jetzig version in the CLI app's `build.zig` and lock it to a GitHub release.

Anyway, Git invocations are all gone now.(